### ISSUE TITLE
[Config Support]:  mount second drive in Home-Assistant to be used as media repository for Frigate

### ISSUE THREAD

**Issue by nmsoares** [blakeblackshear/frigate#4644]
### Describe the problem you are having

How can I mount a second drive (internal or external/usb) to be used by Frigate only ? 

### Version

Home-Assistant 2022.12.0 with Frigate 0.11.1-2EADA21

### Frigate config file

```yaml
mqtt:
  host: 10.0.0.254
  port: 1883
  topic_prefix: frigate
  client_id: frigate 
  user: xxxxxxxxx
  password: 'xxxxxxxxxx'
  stats_interval: 60

##

birdseye:
  # Optional: Enable birdseye view (default: shown below)
  enabled: True
  # Optional: Width of the output resolution (default: shown below)
  width: 1280
  # Optional: Height of the output resolution (default: shown below)
  height: 720

  quality: 8

  mode: continuous


detectors:
  coral1:
    type: edgetpu
    device: usb:0
#    num_threads: 3
  coral2:
    type: edgetpu
    device: usb:1
#    num_threads: 3


record:
  events:
#    max_seconds: 350
    pre_capture: 15
    post_capture: 25
  retain:
    mode: active_objects
    days: 8

ffmpeg:
  global_args: -hide_banner -loglevel warning
  output_args:
    rtmp: -c copy -f flv

cameras:

#################################### Front Camera #######################################
  front_cam_frg:
    record:
      enabled: True

    ffmpeg:

      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer+genpts+discardcorrupt
        - -rtsp_transport
        - tcp
        - -flags
        - low_delay
        - -strict
        - experimental

      hwaccel_args:
        - '-hwaccel'
        - qsv
        - '-qsv_device'
        - /dev/dri/renderD128
        - '-hwaccel_device'
        - '-hwaccel_output_format'

######## RTSP Config
      inputs:
        - path: rtsp://xxxxxxxx:!xxxxxxxx!@10.0.10.192:554//h264Preview_01_sub
          roles:
            - detect

        - path: rtsp://xxxxxxxx:!xxxxxxxx!@10.0.10.192:554//h265Preview_01_main
          roles:
            - record

      output_args:
        record: -f segment -segment_time 30 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy -an #-tag:v hvc1

    rtmp:
      enabled: false

    motion:
      mask:
        - 1280,0,1060,0,1171,317,1280,322
        - 83,296,0,525,0,236
#        - 1280,0,1280,182,1000,137,660,110,340,105,0,142,0,0
        - 1280,0,1280,182,1013,151,657,120,208,142,0,159,0,0
    zones:
      front_out:
        coordinates: 1280,720,1280,381,1185,285,999,266,616,245,208,249,0,295,0,720 #2304,1296,2304,486,1465,418,963,402,0,435,0,1296 #640,480,640,132,449,113,257,109,0,118,0,480
      front_in:
        coordinates: 1280,0,1280,268,637,234,0,248,0,0 #2304,0,2304,494,1182,406,0,446,0,0 #640,0,640,131,272,108,0,125,0,0
      parking_zone:
        coordinates: 67,229,468,228,799,223,1130,247,1016,149,936,88,679,56,556,128,444,65,235,90

    objects:
      track:
        - person
        - dog
        - car
      filters:
        car:
          mask: 1280,0,1280,196,1165,177,965,157,863,90,330,77,193,142,0,155,0,0
          threshold: 0.65
        person:
          mask: 1280,0,1280,199,1082,169,778,144,498,127,181,146,0,154,0,0
          threshold: 0.73

    detect:
      width: 1280 #1024 #1280 #2304 #640
      height: 720 #576 #720 #1296 #360
      fps: 7 #5

#################################### Front Garden Camera #######################################

  front_garden_cam_frg:
    record:
      enabled: True

    ffmpeg:

      hwaccel_args:
        - -hwaccel
        - vaapi
        - -hwaccel_device
        - /dev/dri/renderD128
        - -hwaccel_output_format
        - yuv420p

      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer+genpts+discardcorrupt
        - -flags
        - low_delay
        - -strict
        - experimental
        - -analyzeduration
        - 1000M
        - -probesize
        - 1000M
        - -rw_timeout
        - "5000000"


      inputs:
        - path: http://10.0.10.191/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=xxxxxxxx&password=!xxxxxxxx!
          roles:
            - record
            - rtmp
        - path: http://10.0.10.191/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=xxxxxxxx&password=!xxxxxxxx! 
          roles:
            - detect

      output_args:
        record: -f segment -segment_time 90 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy

    motion:
      mask:
        - 1280,0,1280,125,1149,121,1026,0

    zones:
      front_in:
        coordinates: 0,720,1280,720,1280,339,944,371,790,0,0,0,0,289  #0,720,1280,720,1024,365,660,461,540,0,0,0,0,289

      door_windows:
        coordinates: 954,46,764,110,731,446,865,412,1080,375 # 954,46,764,110,731,446,865,412,1080,375
    objects:
      track:
        - person
        - dog
      filters:
        person:
          threshold: 0.7

    detect:
      stationary:
        interval: 0
        threshold: 60
      width: 1024 #1280 #640
      height: 576 #720 #480
      fps: 6

#################################### Back Pool Camera #######################################

  back_cam_frg:
    record:
      enabled: True

    ffmpeg:

      hwaccel_args:
        - -hwaccel
        - vaapi
        - -hwaccel_device
        - /dev/dri/renderD128
        - -hwaccel_output_format
        - yuv420p

      input_args:
        - -avoid_negative_ts
        - make_zero
        - -fflags
        - nobuffer+genpts+discardcorrupt
        - -flags
        - low_delay
        - -strict
        - experimental
        - -analyzeduration
        - 1000M
        - -probesize
        - 1000M
        - -rw_timeout
        - "5000000"

      inputs:
        - path: http://10.0.10.190/flv?port=1935&app=bcs&stream=channel0_main.bcs&user=xxxxxxxx&password=!xxxxxxxx!
          roles:
            - record
            - rtmp
        - path: http://10.0.10.190/flv?port=1935&app=bcs&stream=channel0_ext.bcs&user=xxxxxxxx&password=!xxxxxxxx!
          roles:
            - detect

      output_args:
        record: -f segment -segment_time 60 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c copy

    motion:
      mask:
        - 540,720,1280,720,1280,276,1280,0,0,0,0,206,0,720,319,720,152,480,60,197,813,42,1245,325

    zones:
      in_pool:
        coordinates: 414,502,944,301,739,165,263,290

      back_garden:
        coordinates: 444,121,793,75,875,103,1211,348,504,720,363,720,184,475,114,356,70,216


    objects:
      filters:
        person:
          mask:
            - 182,480,640,480,640,295,640,0,0,0,0,480,152,480,28,146,406,32,615,209
          threshold: 0.7
        dog:
          mask:
            - 182,480,640,480,640,295,640,0,0,0,0,480,152,480,28,146,406,32,615,209
          threshold: 0.7

    detect:
      stationary:
        interval: 0
        threshold: 40
      width: 1024 #1280 #640
      height: 576 #720 #480
      fps: 6

##########################################################################################
```

### Relevant log output

```shell
n/a
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

USB

**Comment by NickM-27** [comment]
This is covered in the docs https://docs.frigate.video/installation#home-assistant-operating-system-hassos

It's possible but not easy in HA OS and is why we recommend running frigate in docker directly (since it is easy to do outside of HA OS limitations)

**Comment by felipecrs** [comment]
As the link is broken, this is where it pointed:

https://community.home-assistant.io/t/solved-mount-usb-drive-in-hassio-to-be-used-on-the-media-folder-with-udev-customization/258406/46

**Comment by felipecrs** [comment]
Just to document my findings:

1. This fork for the udev rule is the best because it has support for partition labels and the new NTFS driver: https://gist.github.com/microraptor/be170ea642abeb937fc030175ae89c0c

1. Label your external disk partition with `frigate` to ensure it will be mounted under `/media/frigate`
   - I used `tune2fs –L frigate /dev/sdb1` since my file system is ext4.

1. Before mounting anything:
   
   ```bash
   # Ensure there's nothing in the frigate folder already when UNMOUNTED
   rm -rf /mnt/data/supervisor/media/frigate
   mkdir /mnt/data/supervisor/media/frigate
   
   # Lock it to prevent unintentional writes if the external disk is unmounted
   chattr +i /mnt/data/supervisor/media/frigate
   
   # Confirm you can't write to it
   if touch /mnt/data/supervisor/media/frigate/test.txt; then echo FAILED; else echo WORKED; fi
   ```

1. Copy the udev rule as is, without modifications, to your `/etc/udev/rules.d/80-mount-usb-to-media-by-label.rules`:

   ```bash
   curl -fsSL -o /etc/udev/rules.d/80-mount-usb-to-media-by-label.rules https://gist.githubusercontent.com/microraptor/be170ea642abeb937fc030175ae89c0c/raw/3f9d8901df3a63864393b0d5d8b19f939fc819fa/80-mount-usb-to-media-by-label.rules
   ```

Profit! IMO this is much better than using the Samba NAS add-on.

Another tip: you can use [this](https://community.home-assistant.io/t/add-on-hassos-ssh-port-22222-configurator/264109) to enable the 22222 SSH port without needing physical access to your server.

**Comment by Chaoscontrol** [comment]
@felipecrs I have used this method and it works great to mount the drive. However, I have encountered an issue where I cannot move any files into the mounted USB drive (4TB), because Samba thinks the drive is full. It counts the files inside the USB drive as if they were into the main drive (256GB). But what is full is the main drive (256GB), where `/media/` is. The 4tb USB drive is mostly empty. 

<img width="1073" height="462" alt="Image" src="https://github.com/user-attachments/assets/0c92d3ba-442b-4446-a37c-d75d08b13302" />
<img width="549" height="338" alt="Image" src="https://github.com/user-attachments/assets/34b8db81-f7b2-4d0d-9d1e-4eb83addccd6" />

Did no-one else find this problem? 

It's a Samba issue, but essentially makes Samba unusable after doing this.

**Comment by felipecrs** [comment]
I don't know because I don't use Samba.

I know you are not asking for an alternative, but I would suggest [SSHFS-Win](https://github.com/winfsp/sshfs-win) with the [Advanced SSH add-on](https://github.com/hassio-addons/addon-ssh) if you are considering one.

**Comment by Chaoscontrol** [comment]
It's always good to have alternatives. This seems like a good one, thanks! 

For now I've moved back to using Samba NAS (the base version instead of v2 beta) and it seems more stable and solves the issue, as it mounts outside /media/ so I can move files via Samba without it thinking it's full. If it ends up not being reliable I'll def try this. 

Edit: I ended up going back to this method to remove an extra addon point of failure, and an intricate mount to network share method that seemed too complex. I also managed to use SSHFS and it worked perfect to work around SMB limitations with free space. 

Thanks!
